src/Magnum/Platform/EmscriptenApplication.h

```diff
@@ -0,0 +1,1509 @@
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#if defined(CORRADE_TARGET_EMSCRIPTEN) || defined(DOXYGEN_GENERATING_OUTPUT)
+/** @file
+ * @brief Class @ref Magnum::Platform::EmscriptenApplication, macro @ref MAGNUM_EMSCRIPTENAPPLICATION_MAIN()
+ */
+#endif
+
+#include <memory>
+#include <string>
+
+#include <Corrade/Containers/ArrayView.h>
+#include <Corrade/Containers/Pointer.h>
+
+#include "Magnum/Magnum.h"
+#include "Magnum/Tags.h"
+#include "Magnum/GL/GL.h"
+#include "Magnum/Math/Vector4.h"
+#include "Magnum/Platform/Platform.h"
+
+#include <emscripten/html5.h>
+
+#if defined(CORRADE_TARGET_EMSCRIPTEN) || defined(DOXYGEN_GENERATING_OUTPUT)
+struct EmscriptenMouseEvent;
+
+namespace Magnum { namespace Platform {
+
+/** @nosubgrouping
+@brief Emscripten application
+
+Application running on Emscripten.
+
+This application library is available only on
+@ref CORRADE_TARGET_EMSCRIPTEN "Emscripten", see respective sections
+in the @ref building-corrade-cross-emscripten "Corrade" and
+@ref building-cross-emscripten "Magnum" building documentation. It is built if
+`WITH_EMSCRIPTENAPPLICATION` is enabled when building Magnum.
+
+@section Platform-EmscriptenApplication-bootstrap Bootstrap application
+
+Fully contained base application using @ref EmscriptenApplication for Emscripten
+build and CMake setup is available in `base-emscripten` branch of
+[Magnum Bootstrap](https://github.com/mosra/magnum-bootstrap) repository,
+download it as [tar.gz](https://github.com/mosra/magnum-bootstrap/archive/base-emscripten.tar.gz)
+or [zip](https://github.com/mosra/magnum-bootstrap/archive/base-emscripten.zip)
+file. After extracting the downloaded archive, you can build in
+the same way as with @ref Sdl2Application.
+
+Detailed information about deployment for Emscripten and all needed boilerplate
+together with a troubleshooting guide is available in @ref platforms-emscripten.
+
+@section Platform-EmscriptenApplication-usage General usage
+
+Request the `EmscriptenApplication` component of the `Magnum` package and link
+to the `Magnum::EmscriptenApplication` target:
+
+@code{.cmake}
+find_package(Magnum REQUIRED)
+if(CORRADE_TARGET_EMSCRIPTEN)
+    find_package(Magnum REQUIRED EmscriptenApplication)
+endif()
+
+# ...
+if(CORRADE_TARGET_EMSCRIPTEN)
+    target_link_libraries(your-app Magnum::EmscriptenApplication)
+endif()
+@endcode
+
+If no other application is requested, you can also use the generic
+`Magnum::Application` alias to simplify porting. Again, see @ref building and
+@ref cmake for more information. Note that unlike on other platforms you need
+to create *shared library* instead of executable.
```

Comment:
Ummm ... a copypaste leftover from `AndroidApplication`? :sweat_smile:  Similarly for the JNI blurb below.